### PR TITLE
Some docs work to prepare for 5.0

### PR DESCRIPTION
**Pull request by takluyver**
- Improve release notes
- Simplify docs on terminal colours
- Add info on customising prompts
- Update docs on integrating event loops

This also adds back an interface to add input hooks for other event loops beyond those we define.

### CODE REVIEW

**Comment by minrk** [comment]
Very nice. cc @willingc for docs/release review, but 👍 from me.

**Comment by takluyver** [comment]
I've also reverted part of #9567 - I hadn't had time to look closely at it before, but I prefer having separate pages for the kernel and terminal options rather than one big page.

File: docs/source/whatsnew/version5.rst [hunk 1]
```diff
@@ -5,71 +5,69 @@
...
+  event. See :ghissue:`9447` and :ghpull:`9453`.
+- Integration with pydb has been removed since pydb development has been stopped
+  since 2012, and pydb is not installable from PyPI.
+- The ``autoedit_syntax`` option has apparently been broken for many years.
+  It has been removed.
+
+New terminal interface
+~~~~~~~~~~~~~~~~~~~~~~
+
+The overhaul of the terminal interface will probably cause a range of minor issues;
```
Comment (willingc) [inline docs/source/whatsnew/version5.rst]: Really like this section for the user 👍 

Suggested text, but fine as is too.

The terminal interface's overhaul (or update) may cause a range of minor issues for existing users. This is inevitable for such a significant change, and we've done our best to minimize these issues. Some issues that we are aware of as well as some suggestions for the user:

File: docs/source/config/details.rst [hunk 2]
```diff
@@ -2,89 +2,85 @@
...
+Custom Prompts
+==============
+
+.. versionchanged:: 5.0
+
+From IPython 5, prompts are produced as a list of Pygments tokens, which are
+tuples of (token_type, text). You can customise prompts by writing a method
+which generates a list of tokens.
+
+There are four kinds of prompt:
```
Comment (willingc) [inline docs/source/config/details.rst]: Not sure if this should be prompt or prompts. Likely could be either. No need to revise if you are fine with prompt.

**Comment by willingc** [comment]
@takluyver Nice!

A couple of minor comments. +1 to merging. Someone with permissions on the repo will need to do the merge.

File: docs/source/whatsnew/version5.rst (hunk 1 above)
Comment (takluyver) [inline docs/source/whatsnew/version5.rst]: Reworded somewhere between my text and your suggestion.

File: docs/source/config/details.rst (hunk 2 above)
Comment (takluyver) [inline docs/source/config/details.rst]: I think it could be either - it sounds slightly better in my head as is.

**Pull request merged**